src/lib/history.js

```diff
@@ -1,6 +1,17 @@
 let getLocation = source => {
+  // ensure key properties are copied to location
+  // see issue #252
+  const location = { ...source.location };
```

Comment:
doesn't this still copy everything over?

Reply:
I was thinking about keeping the original logic, first copy all own properties using spread operator, then ensure the three key properties are copied.